.github/workflows/dependabot-auto-merge.yml

```diff
@@ -1,14 +1,28 @@
-name: dependabot-auto-merge
+name: Dependabot auto-approve and auto-merge
+on: pull_request
 
-on:
-  pull_request:
+permissions:
+  contents: write
+  pull-requests: write
 
 jobs:
-  auto-merge:
+  dependabot:
     runs-on: ubuntu-latest
+    if: github.actor == 'dependabot[bot]'
     steps:
-      - uses: actions/checkout@v4
-      - uses: ahmadnassri/action-dependabot-auto-merge@v2
+      - name: Dependabot metadata
+        id: metadata
+        uses: dependabot/fetch-metadata@v1
         with:
-          target: minor
-          github-token: ${{ secrets.DEPENDABOT_GITHUB_TOKEN }}
\ No newline at end of file
+          github-token: "${{ secrets.GITHUB_TOKEN }}"
+      - name: Approve a PR (patch and minor updates only)
```

Comment:
We're going to have to see how this interacts with the code owner approval requirement -- I'd like to merge and check what our options are after that, but I don't really expect this to work as-is.